### DISCUSSION TITLE
[Question] upper half circle

### DISCUSSION THREAD

**Discussion by racuca** [flet-dev/flet#1874] [Q&A]
### Question

I can create a bowl-like bottom semicircle on canvas as code. However, it is difficult to create a rainbow-like upper semicircle. How can I make it?

### Code sample

```python
import math

import flet as ft
import flet.canvas as cv

def main(page: ft.Page):
    stroke_paint = ft.Paint(stroke_width=2, style=ft.PaintingStyle.STROKE)
    fill_paint = ft.Paint(style=ft.PaintingStyle.FILL)
    cp = cv.Canvas(
        [
            cv.Arc(65, 65, 70, 70, 0, 3.14, paint=stroke_paint),
            cv.Arc(50, 50, 100, 100, 0, 3.14, paint=stroke_paint),
        ],
        width=float("inf"),
        expand=True,
    )

    page.add(cp)

ft.app(main)
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
Can you show an image of what you wish to achieve?

**Reply by racuca** [reply]
Sorry. I am late. I want it like this finally. 

![image](https://github.com/flet-dev/flet/assets/3173661/cfb96a3e-9507-4bc7-94bd-a1194d71406b)
